hooks/set-k8s-creds.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+ONDEMAND_USERNAME="$1"
+if [ "x${ONDEMAND_USERNAME}" = "x" ]; then
+  echo "Must specify username"
+  exit 1
+fi
+
+# This file has environment specific stuff like TOKEN_URL, CLIENT_SECRET,
+# and K8S_USERNAME
+# shellcheck disable=SC1091
+source "/opt/osc/etc/ondemand-k8.env"
+
+# we use pass ACCESS_TOKEN into the id-token arg. That's OK, it works and refreshes.
+sudo -u "$ONDEMAND_USERNAME" kubectl config set-credentials "$K8S_USERNAME" \
+   --auth-provider=oidc \
+   --auth-provider-arg=idp-issuer-url="$TOKEN_URL" \
+   --auth-provider-arg=client-id="ondemand" \
```

Comment:
That client ID is not going to work.  The client ID for each OnDemand instance is the same as the `servername` in OOD configs so for dev it's `ondemand-dev.osc.edu`. I think that is one thing we will have to put into `/opt/osc/etc/ondemand-k8.env`

Reply:
```suggestion
   --auth-provider-arg=client-id="$CLIENT_ID" \
```

Reply:
Yep. added.